ai-ml/trainium-inferentia/examples/ray-serve/stable-diffusion-inf2/ray-service-stablediffusion.yaml

```diff
@@ -0,0 +1,128 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: stablediffusion
+
+---
+apiVersion: ray.io/v1alpha1
+kind: RayService
+metadata:
+  name: stablediffusion-service
+  namespace: stablediffusion
+spec:
+  serviceUnhealthySecondThreshold: 900
+  deploymentUnhealthySecondThreshold: 300
+  serveConfig:
+    importPath: ray_serve_stablediffusion:entrypoint # Specify the correct path to your Python script
+    runtimeEnv: |
+      env_vars: {"MODEL_ID": "aws-neuron/stable-diffusion-xl-base-1-0-1024x1024"}  # Replace with the appropriate model ID
+
+  rayClusterConfig:
+    rayVersion: '2.7.1'
+    headGroupSpec:
+      serviceType: NodePort
+      headService:
+        metadata:
+          name: stablediffusion-service
+          namespace: stablediffusion
+      rayStartParams:
+        dashboard-host: '0.0.0.0'
+      template:
+        spec:
+          containers:
+          - name: ray-head
+            image: public.ecr.aws/r8y1i9f8/ray-serve-stable-diffusion:latest # Image created using the Dockerfile attached in the folder
+            imagePullPolicy: Always # Ensure the image is always pulled when updated
+            lifecycle:
+              preStop:
+                exec:
+                  command: ["/bin/sh", "-c", "ray stop"]
+            ports:
+            - containerPort: 6379
+              name: gcs
+            - containerPort: 8265
+              name: dashboard
+            - containerPort: 10001
+              name: client
+            - containerPort: 8000
+              name: serve
+            volumeMounts:
+            - mountPath: /tmp/ray
+              name: ray-logs
+            resources:
+              limits:
+                cpu: "2"
+                memory: "8G"
+              requests:
+                cpu: "2"
+                memory: "8G"
+          nodeSelector:
+            workload: "rayhead"
+          volumes:
+          - name: ray-logs
+            emptyDir: {}
+
+    workerGroupSpecs:
+    - groupName: inf2-worker-group
+      replicas: 1
+      minReplicas: 1
+      maxReplicas: 1
+      rayStartParams: {}
+      template:
+        spec:
+          containers:
+          - name: ray-worker
+            image: public.ecr.aws/r8y1i9f8/ray-serve-stable-diffusion:latest # Image created using the Dockerfile attached in the folder
+            imagePullPolicy: Always # Ensure the image is always pulled when updated
+            lifecycle:
+              preStop:
+                exec:
+                  command: ["/bin/sh", "-c", "ray stop"]
+            resources:
+              limits:
+                cpu: "90"
+                memory: "360G"
```

Comment:
These requests and limits seem to be off
1. What is consuming 360G of host memory? Typically with these workloads, the data all reside within HBM on the accelerators 
2. What is the intended pod to accelerator/device ratio? Is it one pod per host/instance, and that pod consumes all available resources - or is it multiple pods? This will also determine whether or not you need the collective communications library in the container (if its one pod on one host consuming all neuron devices then you do not need the collective communication library)

Reply:
This was primarily based on limits supported by Inf2 instances. Will adjust these based on further testing.

Reply:
I think you are over-constraining - if your main requirement is 2 neuron cores, request that until you need to specify other requirements. right now you are requesting 2 neuron cores and almost all of the host CPU/memory so you won't be able to schedule any more pods on the instance like this